src/components/Select.vue

```diff
@@ -596,6 +607,8 @@
             option = this.selectedValue.concat(option)
           }
           this.updateValue(option);
+        } else if (this.deselectByOption && (this.clearable || this.multiple && this.val.length > 1)) {
```

Comment:
@andreasvirkus @sagalbot I think this feature makes more sense in `multiple` (and optionally with `closeOnSelect=false`). 

One use case would be allowing users to freely select/deselect multiple items while the dropdown is open. In our web app it's the only way to deselect the items, since we only show a certain number of selected items and we do not have the (X) button next to each of them (our dropdown toggle box does not expand to fit all selected items).

What is still needed for this PR to be merged?

Reply:
@codynguyen that's exactly our use case as well - a user can **only** deselect options directly from the dropdown and not actually from the input.

I'm not sure what's needed for this to be merged. I think Jeff (@sagalbot) is currently working on supporting Vue 3?